### DISCUSSION TITLE
Partitioner System

### DISCUSSION THREAD

**Discussion by lukuna** [idaholab/moose#21207] [Q&A General] [answered]
Hi all,
I recently found the documentation here: https://mooseframework.inl.gov/syntax/Mesh/Partitioner/index.html to be very interesting and potentially useful for some calculations I am running on HPC.  But I am curious what is the default Partitioner that MOOSE uses?  Is it just GridPartitioner?  

On the same note, if I were to leave my executioner block as:

```
[Executioner]
  type = Transient
  solve_type = 'NEWTON'
  start_time = 0.0
  end_time = 2800
[]
```

for, lets say, a simple TensorMechanics simulation, what are the default solver options?  Are there any PETSc options being used?

Thanks,
Lukasz

**Comment by loganharbour** [comment, marked as answer]
MOOSE by default uses metis for partitioning. With parallel runs you can see the partitioner used in the header:

```
Mesh:
  Parallel Type:           replicated
  Mesh Dimension:          2
  Spatial Dimension:       2
  Nodes:
    Total:                 121
    Local:                 67
    Min/Max/Avg:           54/67/60
  Elems:
    Total:                 100
    Local:                 50
    Min/Max/Avg:           50/50/50
  Num Subdomains:          1
  Num Partitions:          2
  Partitioner:             metis
```

Similarly, you can see a generic solver summary:

```
Execution Information:
  Executioner:             Steady
  Solver Mode:             Preconditioned JFNK
  PETSc Preconditioner:    hypre boomeramg
```

**Reply by lukuna** [reply]
Thats pretty interesting.  My job prints the following under Execution:

```
Execution Information:
  Executioner:             Transient
  TimeStepper:             FunctionDTGrowth
  Solver Mode:             NEWTON
  MOOSE Preconditioner:    SMP
```

Even if I add 
```
  petsc_options_iname = '-pc_hypre_type'
  petsc_options_value = 'boomerang'
```

it doesn't change that execution block (meaning I Don't see the PETSc options).  Though, I do see where the partitioner information is now.

**Reply by loganharbour** [reply]
I would rely on querying PETSc to see what options are used. I typically use `-options_view` but I'm sure there's something better:

```
-mat_mffd_type wp
-options_view
-PC_HYPRE_TYPE boomeramg
-PC_TYPE hypre
-snes_converged_reason ::failed
-snes_mf_operator
```

is some example output
